bus.py

```diff
@@ -16,14 +16,11 @@ def _road_point_generator(roads):
         length_sum = 0.0
         found_road = False
         for road in roads:
-            from_node = road.getFromNode().getCoord()
-            to_node = road.getToNode().getCoord()
-
             if (length_sum + road.getLength()) >= distance:
                 # Compute the exact point on the selected road
 
                 # Distance along the road segment
-                remaining = length_sum - distance
+                remaining = distance - length_sum
```

Comment:
This is wrong, `length_sum` should always be bigger than distance. See my fix in #33

Reply:
But it's not in this branch of the if-statement. That's why you got an error.